crates/router/src/core/payments/helpers.rs

```diff
@@ -1821,16 +1825,60 @@ pub async fn retrieve_card_with_permanent_token(
         })?;
 
     if !business_profile.is_network_tokenization_enabled {
```

Comment:
in `profile` if the `is_network_tokenization_enabled` is false then only the if block will get executed.

Is the below implementation of network_tokenization is in right place?

Reply:
> in `profile` if the `is_network_tokenization_enabled` is false then only the if block will get executed.
> 
> Is the below implementation of network_tokenization is in right place?

Yes, it is in the right place.
This change is required when we do the MIT with the card details stored in locker and network transaction id stored in the payment methods table (it will be populated in the `mandate_reference_id`). So in this case the card details retrieved from locker should be used to construct `CardDetailsForNetworkTransactionId` as we will not have cvc.